### PR TITLE
Add tests for tar

### PR DESCRIPTION
**Pull request by bertold**
* there is no library for compressing using the traditional Z format, so I added a test file for this

* added a new dependency for writing bzip2 

* there was also a typo in the error message in tar.go that I fixed

### CODE REVIEW

**Comment by davidnewhall** [comment]
Thanks for this!

If you need to pull in another library to compress files, I'd rather just put them in the repo. I'm not against that, but when you can use the stdlib or already-available open source package we use, go ahead and create the files. The tests also run on Windows, so os.exec is going to be a problem. Again, just create some test files and include them.

File: tar.go [hunk 1]
```diff
@@ -62,7 +62,7 @@ func ExtractTarZ(xFile *XFile) (int64, []string, error) {
 
 	zipStream, err := lzw.NewReader(compressedFile)
 	if err != nil {
-		return 0, nil, fmt.Errorf("xz.NewReader: %w", err)
+		return 0, nil, fmt.Errorf("z.NewReader: %w", err)
```
Comment (davidnewhall) [inline tar.go:65]: `lzw.NewReader` ?

**Review by davidnewhall — APPROVED** [review]
eh, I'm actually okay with this. Thanks again!

**Comment by bertold** [comment]
Thank you for the invite to collaborate - I am humbled.  Let me know @davidnewhall if it is OK to merge.

**Review by davidnewhall — APPROVED** [review]
OK to merge. Thanks!

**Pull request merged**